tests/ext/test_applicationbuilder.py

```diff
@@ -379,6 +416,36 @@ class Client:
         assert client.http1 is True
         assert client.http2 is False
 
+    def test_custom_socket_options(self, builder, monkeypatch, bot):
+        httpx_request_kwargs = []
+        httpx_request_init = HTTPXRequest.__init__
+
+        def init_transport(*args, **kwargs):
+            nonlocal httpx_request_kwargs
+            # This is called once for request and once for get_updates_request, so we make
+            # it a list
+            httpx_request_kwargs.append(kwargs.copy())
+            httpx_request_init(*args, **kwargs)
+
+        monkeypatch.setattr(HTTPXRequest, "__init__", init_transport)
+
+        builder.token(bot.token).build()
+        assert httpx_request_kwargs[0].get("socket_options") is None
+        assert httpx_request_kwargs[1].get("socket_options") is None
+
+        httpx_request_kwargs = []
+        ApplicationBuilder().token(bot.token).socket_options(((1, 2, 3),)).connection_pool_size(
+            "request"
+        ).get_updates_socket_options(((4, 5, 6),)).get_updates_connection_pool_size(
+            "get_updates"
+        ).build()
+
+        for kwargs in httpx_request_kwargs:
+            if kwargs.get("connection_pool_size") == "request":
+                assert kwargs.get("socket_options") == ((1, 2, 3),)
+            else:
+                assert kwargs.get("socket_options") == ((4, 5, 6),)
```

Comment:
```suggestion
                assert kwargs.get("get_updates_socket_options") == ((4, 5, 6),)
```
shouldn't it be this?

Reply:
oh nvm it is kwargs from HTTPXRequest, that parameter doesn't exist